diff --git a/console-files/configure.sh b/console-files/configure.sh
index 31202c7..cfc3611 100755
--- a/console-files/configure.sh
+++ b/console-files/configure.sh
@@ -64,4 +64,18 @@ if [[ -z "$(cat $sshdir/known_hosts | grep qcYjThFg)" ]]; then
     cat $sshdir/known_hosts
 fi
 
+
+# power button override
+if [[ ! -f /etc/acpi/events/powerbtn_override ]]; then
+    echo "enabling acpid"
+    sudo systemctl enable acpid
+    echo "copying power button override scripts and config"
+    cp $rqn/login.conf /etc/systemd/
+    cp $rqn/powerbtn_override /etc/acpi/events/
+    cp $rqn/powerbtn_handler.sh /etc/acpi/
+    cp $rqn/toggle_ip_qr.ip $base/
+    echo "restarting acpid"
+    sudo systemctl restart acpid
+fi
+
 echo "Done configuring"
diff --git a/console-files/install.sh b/console-files/install.sh
index 4e5b477..e13c0bd 100755
--- a/console-files/install.sh
+++ b/console-files/install.sh
@@ -15,6 +15,18 @@ if ! command -v xdotool >/dev/null; then
     echo "installing xdotool"
     sudo apt install -y xdotool
 fi
+if ! command -v pqiv >/dev/null; then
+    echo "installing pqiv"
+    sudo apt install -y pqiv
+fi
+if ! command -v wmctrl >/dev/null; then
+    echo "installing wmctrl"
+    sudo apt install -y wmctrl
+fi
+if ! command -v acpid >/dev/null; then
+    echo "installing acpid"
+    sudo apt install -y acpid
+fi
 if ! command -v tmux >/dev/null; then
     echo "installing tmux"
     sudo apt install -y tmux
diff --git a/console-files/login.conf b/console-files/login.conf
new file mode 100644
index 0000000..cc9eb07
--- /dev/null
+++ b/console-files/login.conf
@@ -0,0 +1,4 @@
+# Note: there was a license notice here
+
+[Login]
+HandlePowerKey=ignore
diff --git a/console-files/powerbtn_handler.sh b/console-files/powerbtn_handler.sh
new file mode 100755
index 0000000..e41ee00
--- /dev/null
+++ b/console-files/powerbtn_handler.sh
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+
+DEBOUNCE_FILE="/tmp/powerbtn_debounce"
+DEBOUNCE_TIME=2
+CURRENT_TIME=$(date +%s)
+
+
+if [ -f "$DEBOUNCE_FILE" ]; then
+    LAST_TIME=$(cat $DEBOUNCE_FILE)
+    DIFF_TIME=$((CURRENT_TIME - LAST_TIME))
+    if [ "$DIFF_TIME" -lt "$DEBOUNCE_TIME" ]; then
+	exit 0
+    fi
+fi
+
+
+rm $DEBOUNCE_FILE
+echo $CURRENT_TIME > $DEBOUNCE_FILE
+
+# toggle qr image overlay
+sudo -u requin /home/requin/toggle_ip_qr.sh > /home/requin/logs/qr_ip.out 2>&1
+
diff --git a/console-files/powerbtn_override b/console-files/powerbtn_override
new file mode 100644
index 0000000..4a8c5b3
--- /dev/null
+++ b/console-files/powerbtn_override
@@ -0,0 +1,2 @@
+event=button/power.*LNXPWRBN
+action=/etc/acpi/powerbtn_handler.sh > /home/requin/logs/powerbtn.out
\ No newline at end of file
diff --git a/console-files/setup.sh b/console-files/setup.sh
index 57a9701..991bf1d 100755
--- a/console-files/setup.sh
+++ b/console-files/setup.sh
@@ -53,6 +53,8 @@ systemctl disable lightdm.service
 apt install -y curl git
 apt install -y qrencode 
 apt install -y xdotool
+apt install -y acpid
+apt install -y pqiv wmctrl
 
 # install convenience tools
 apt install -y emacs
@@ -88,6 +90,9 @@ systemctl enable getty@tty1.service
 cp $dest/.xinitrc $base/
 cp $dest/.bashrc $base/
 
+# other configuration (power button override)
+$dest/configure.sh
+
 # done with setup
 echo "Done."
 sleep 3
diff --git a/console-files/toggle_ip_qr.sh b/console-files/toggle_ip_qr.sh
new file mode 100755
index 0000000..e74512d
--- /dev/null
+++ b/console-files/toggle_ip_qr.sh
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+## dependencies
+# if ! dpkg -l pqiv 2>/dev/null | grep . -q; then
+#     sudo apt install pqiv
+# fi
+# if ! dpkg -l xdotool 2>/dev/null | grep . -q; then
+#     sudo apt install xdotool
+# fi
+# if ! dpkg -l wmctrl 2>/dev/null | grep . -q; then
+#     sudo apt install wmctrl
+# fi
+
+export DISPLAY=:0
+
+## kill current QR overlay if it already exists
+if pgrep -f pqiv_qr >/dev/null; then
+    pkill -f pqiv_qr
+    exit
+fi
+
+## make the QR image from local IP
+ip addr | grep 'inet ' | grep -vw lo | tail -1 | sed 's/.*inet \([^ \/]*\).*/http:\/\/\1:3000/' | \
+    qrencode  -s 10  -m 1  --foreground=000000AA --background=FFFFFFBB  -o /home/requin/qr.png
+
+
+## display it with an overlay
+pqiv -c -i -T pqiv_qr /home/requin/qr.png &
+# -c means tranparent background / no borders
+# -i means do not display info text
+# -T sets the title for the image
+
+
+#######
+
+#sleep 1
+
+## make it stay on top of other windows
+#wmctrl -i -r $winID -b add,above
+
